url43a0865bb0_kevinjdolan_darwin_tgz-pJ8-tui_DriverJ8/src/darwInvest/data/gathering/dataSource/ReutersGoogleFinanceNYSE.java

```diff
@@ -0,0 +1,192 @@
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.Scanner;
+
+import darwInvest.data.NewsEvent;
+import darwInvest.data.Ticker;
+import darwInvest.data.gathering.dataSource.reuters.URLExploreJob;
+import darwInvest.data.gathering.dataSource.reuters.URLExploreJobInput;
+import darwInvest.data.gathering.dataSource.reuters.URLExploreJobOutput;
+import darwInvest.data.utility.DefaultSerializer;
+import darwInvest.data.utility.Serializer;
+import darwin.work.BusyException;
+import darwin.work.Job;
+import darwin.work.JobRunner;
+import darwin.work.threadPool.ThreadPool;
+
+public class ReutersGoogleFinanceNYSE extends DataSource {
+	private String source;
+	private int nostories;
+	
+	public ReutersGoogleFinanceNYSE(Serializer serializer, Date startdate, Date enddate, int numberStories) {
+		super("ReutersGoogleFinanceNYSE", serializer, startdate, enddate);
+		nostories = numberStories;
+	}
+	
+	public void gatherData() {
+		File target = new File("StockData/" + source);
+		target.mkdir();
+		File listoftickers = new File("RawData/" + super.getName() + "/tickers.txt");
+		Scanner s = null;
+		try {
+			s = new Scanner(new BufferedReader(new FileReader(listoftickers)));
+			s.useDelimiter("[\n\r]+");
+			GoogleFinanceManual csvreader = new GoogleFinanceManual(new DefaultSerializer());
+			while(s.hasNext()) {
+				String currentstock = s.next();
+				Ticker ticker = new Ticker(currentstock);
+				String companyname = ticker.lookupCompanyname();
+				String reuters = "http://www.reuters.com/search?blob=%22" + webize(companyname) +  "%22";
+				//(new Thread(new ParallelReader(ticker,reuters,nostories,startdate,enddate))).start();
+				readInReuters(ticker, reuters, nostories);
+				String companyid = ticker.lookupGooglecompanyid();
+				GregorianCalendar cal = new GregorianCalendar();
+				cal.setTime(startdate);
+				String startmonth = csvreader.MonthtoGoogle(cal);
+				String startday = String.valueOf(cal.get(Calendar.DAY_OF_MONTH));
+				String startyear = String.valueOf(cal.get(Calendar.YEAR));
+				cal.setTime(enddate);
+				String endmonth = csvreader.MonthtoGoogle(cal);
+				String endday = String.valueOf(cal.get(Calendar.DAY_OF_MONTH));
+				String endyear = String.valueOf(cal.get(Calendar.YEAR));
+				String longurl = "http://www.google.com/finance/historical?cid=" + companyid + "&startdate=" 
+				+ startmonth + "+" + startday + "%2C+" + startyear + "&enddate=" + endmonth + "+" + endday + "%2C+"
+				+ endyear + "&output=csv";
+				URL googlecsv = new URL(longurl);
+				System.out.println(currentstock + ":" + longurl);
+				BufferedReader googlecsvread = new BufferedReader(new InputStreamReader(googlecsv.openStream()));
+				ticker.addData(csvreader.readIn(googlecsvread));
+				if (googlecsvread != null) {
+					googlecsvread.close();
+				}
+				addTicker(ticker);
+			}
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
+		} catch (MalformedURLException e) {
+			e.printStackTrace();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This code contains a potential information leakage in the handling of the following *exception logging*: (MalformedURLException e). Stack trace information reveals variable names, method names, line numbers, and other parts of the code.

**Fix**
Release data more selectively, or declassify the data before it&#39;s released. 

**More info**
[Learn more about information leakage](https://cwe.mitre.org/data/definitions/497.html) (external link).